streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/CombinedKeySchema.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals.foreignkeyjoin;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.processor.ProcessorContext;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Factory for creating CombinedKey serializers / deserializers.
+ */
+public class CombinedKeySchema<KO, K> {
```

Comment:
This is the replacement for the CombinedKeySerde. Note that it has much less boilerplate because it's not an actual serde anymore. Also, the "dummy topic" to use is now in the constructor, so it's configured in one place and we don't have to worry about coordinating to use the same topic name everywhere the serdes are used.

Reply:
You have cut the Gordian knot.

Reply:
Hah! Always felt that cutting the knot was cheating...